subtests/docker_cli/deferred_deletion/deferred_deletion.py

```diff
@@ -9,25 +9,37 @@
 
   1) kernel is 3.10.0-632 or above; and
   2) The sysctl knob /proc/sys/fs/may_detach_mounts exists and is 1; and
-  3) dockerd unit file DOES NOT include MountFlags=slave; and
-  4) docker daemon is configured to use devicemapper storage; and
-  5) docker daemon is run with --storage-opts dm.use_deferred_deletion=true
-
-As a precondition for running this test, we check only conditions 4 & 5,
-aborting with TestNAError if either is false. We trust docker-storage-setup
-to only enable #5 if all preconditions are met. Anyone running a system
-with 4 & 5 true but 1, 2, or 3 false deserves what they get.
+  3) docker daemon is configured to use devicemapper storage; and
+  4) docker daemon is run with --storage-opts dm.use_deferred_deletion=true
+
+As a precondition for running this test we check conditions 2, 3 & 4,
+aborting with TestNAError if any is false. We would like to trust
+docker-storage-setup to only enable #4 if all preconditions are met,
+but we can't: RHEL 7.2 shipped with a d-s-s that blindly enabled
+deferred_deletion even though it didn't actually work.
+
+Note that we need to do all the important work from inside a
+helper script, because our autotest process will only have access
+to the container's mounts if MountFlags=slave is *absent* from
+the docker systemd unit file; by default that option is present,
+and it is not clear if/when it will be removed. To make sure
+we have access to the container's mounts we need to nsenter
+the docker daemon's mount namespace, and we can't do that within
+autotest (using ctypes.CDLL and setns) because autotest is
+multithreaded and setns() doesn't allow that.
 
 Operational Summary
 ----------------------
 
 #. Run a docker container
-#. Find out the local (host) mountpoint of its root filesystem. cd into it.
-#. Read the Deferred-Deletion count from 'docker info'.
-#. Remove the container.
-#. Read the Deferred-Deletion count; confirm that it has grown by one.
-#. cd back out of the container rootfs
-#. Read the Deferred-Deletion count; confirm that it has gone back down.
+#. Sigh. Run a test script inside docker mount space that will:
```

Comment:
No need to "Sigh" in documentation 😆   People probably won't know what you're upset about in that context.  As for the test, also no need, this is a perfectly valid strategy.  It also provides a good model for doing these kinds of tricky things for testing other aspects.